### PR TITLE
fix(sbb-calendar): using keyboard navigation loses focus

### LINKED ISSUE #2300: bug(sbb-datepicker): using keyboard navigation loses focus
When opening year / month selection and you select a year / month by keyboard, the focus is lost. you have to press tab a few times to select the month or date.
I think the focus should stay on months / dates.

You can reproduce it here: https://lyne-storybook.app.sbb.ch/?path=/story/components-sbb-datepicker-sbb-datepicker--in-form-field

### PR DESCRIPTION
**Pull request by MarioCastigliano**
## Preflight Checklist

- [x] I have read the [Contributing Guidelines](https://github.com/lyne-design-system/lyne-components/blob/main/CONTRIBUTING.md) for this project.
- [x] I agree to follow the [Code of Conduct](https://github.com/lyne-design-system/lyne-components/blob/main/CODE_OF_CONDUCT.md) that this project adheres to.
- [x] I have searched the [pull request tracker](https://github.com/lyne-design-system/lyne-components/pulls) for a Pull Request (PR) that matches the one I want to submit, without success.

## Issue

This PR Closes #2300 

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes / features)

See [Review Guidelines](../REVIEW.md) for more information what is checked during review process.

## Changes

Changes in this pull request:

- Fix focus loss during keyboard navigation
- Fix data-now="" being incorrectly interpreted as "Jan 1, 1970"

## Browsers

I tested the build on the following browsers:

- [x] Firefox Desktop
- [x] Chrome Desktop
- [ ] Edge Desktop
- [x] Safari Desktop
- [ ] Chrome Mobile
- [ ] Safari Mobile

## Screen readers

I tested the build on the following browsers:

- [ ] JAWS Firefox Desktop
- [ ] JAWS Chrome Desktop
- [ ] NVDA Firefox Desktop
- [ ] NVDA Chrome Desktop
- [ ] VoiceOver Safari Desktop
- [ ] VoiceOver Chrome Desktop
- [ ] VoiceOver Safari Mobile
- [ ] Android Accessibility Suite Chrome Mobile

## Pull request type

Please check the type of change your PR introduces:

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Review by TomMenga — CHANGES_REQUESTED** [review]
It looks legit and good to me.
I just have a doubt on the `.?query...` thing

File: src/components/calendar/calendar.ts [hunk 1]
```diff
@@ -553,7 +552,7 @@ export class SbbCalendarElement extends LitElement {
 
   private _setTabIndex(): void {
     Array.from(
-      this.shadowRoot!.querySelectorAll('.sbb-calendar__cell[tabindex="0"]') ?? [],
+      this.shadowRoot!.querySelectorAll?.('.sbb-calendar__cell[tabindex="0"]') ?? [],
```
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/components/calendar/calendar.ts:555]: Here it's not needed as settabindex is called in updated() method and some not initial calls (like blur).
see https://lit.dev/docs/ssr/authoring/

File: src/components/calendar/calendar.ts [hunk 2]
```diff
@@ -564,16 +563,16 @@ export class SbbCalendarElement extends LitElement {
...
           this._activeDate.getMonth() + 1
         } ${this._activeDate.getFullYear()}"]`,
       ) ??
-      this.shadowRoot!.querySelector(`[data-month="${this._activeDate.getMonth()}"]`) ??
-      this.shadowRoot!.querySelector(`[data-year="${this._activeDate.getFullYear()}"]`);
+      this.shadowRoot!.querySelector?.(`[data-month="${this._activeDate.getMonth()}"]`) ??
+      this.shadowRoot!.querySelector?.(`[data-year="${this._activeDate.getFullYear()}"]`);
     if (!firstFocusable || (firstFocusable as HTMLButtonElement)?.disabled) {
-      firstFocusable = this.shadowRoot!.querySelector('.sbb-calendar__cell:not([disabled])');
+      firstFocusable = this.shadowRoot!.querySelector?.('.sbb-calendar__cell:not([disabled])');
```
[the remark this answers is not in the document]
Comment (jeripeierSBB) [reply src/components/calendar/calendar.ts:575]: also here it seems not to get called on SSR side.
on the other side, it's also not that bad to just have it. For me both would be ok :)

File: src/components/calendar/calendar.e2e.ts [hunk 3]
```diff
@@ -11,6 +11,11 @@ import { SbbCalendarElement } from './calendar';
 describe('sbb-calendar', () => {
   const selected = new Date(2023, 0, 15).getTime() / 1000;
   let element: SbbCalendarElement;
+  const waitForTransition = async (): Promise<void> => {
+    await waitForCondition(() => element.hasAttribute('data-transition'));
+    await waitForLitRender(element);
+    await waitForCondition(() => !element.hasAttribute('data-transition'));
+  };
```
Comment (jeripeierSBB) [inline src/components/calendar/calendar.e2e.ts:18]: I have some doubts that this runs stable, as adding and removing of data-transition could be done faster than the waitforcondition waits.

Comment (jeripeierSBB) [reply src/components/calendar/calendar.e2e.ts:18]: could it be that disableAnimation property is missing on the calender?

**Review by jeripeierSBB — APPROVED** [review]
LGTM

**Pull request merged**